fedora_messaging/message.py

```diff
@@ -288,18 +340,18 @@ def validate(self):
             jsonschema.SchemaError: If either the message header schema or the message body
                 schema are invalid.
         """
-        _log.debug(
-            'Validating message headers "%r" with schema "%r"',
-            self._headers,
-            self.headers_schema,
-        )
-        jsonschema.validate(self._headers, self.headers_schema)
-        _log.debug(
-            'Validating message body "%r" with schema "%r"',
-            self._body,
-            self.body_schema,
-        )
-        jsonschema.validate(self._body, self.body_schema)
+        for schema in (self.headers_schema, Message.headers_schema):
+            _log.debug(
+                'Validating message headers "%r" with schema "%r"',
+                self._headers,
+                schema,
+            )
+            jsonschema.validate(self._headers, schema)
+        for schema in (self.body_schema, Message.body_schema):
+            _log.debug(
+                'Validating message body "%r" with schema "%r"', self._body, schema
+            )
+            jsonschema.validate(self._body, schema)
```

Comment:
You're doing this so people don't override the base schemas too much? Are you thinking of a specific use case?

Reply:
I was thinking that for the headers, it'd be good to validate we added the expected schema, timestamp, etc. keys and the easiest way would be to add it to the schema, but then I wanted people to have the ability to expand upon it. With the body I think it's reasonable to demand all messages be an object, so this makes it enforced.

I can't really think of a use-case for users to actually add headers at the moment so maybe that should all be an internal thing, though.